### PR TITLE
simplify developing modules

### LINKED ISSUE #77621: module tempfile.py causing circular import when following module hacking docs
### Summary

I am trying to run my_test.py by following the documentation in [1].   This fails with this error:

```
╰─➤  python -m ansible.modules.my_test   /tmp/args.json
Traceback (most recent call last):
  File "/usr/lib/python3.8/[runpy.py](http://runpy.py/)", line 194, in \_run\_module\_as\_main
    return \_run\_code(code, main\_globals, None,
  File "/usr/lib/python3.8/[runpy.py](http://runpy.py/)", line 87, in \_run\_code
    exec(code, run\_globals)
  File "/home/dohlemacher/cloud/ansible/lib/ansible/modules/[david.py](http://david.py/)", line 72, in \<module>
    from ansible.module\_utils.basic import AnsibleModule
  File "/home/dohlemacher/cloud/ansible/lib/ansible/module\_utils/[basic.py](http://basic.py/)", line 53, in \<module>
    import tempfile
  File "/home/dohlemacher/cloud/ansible/lib/ansible/modules/[tempfile.py](http://tempfile.py/)", line 79, in \<module>
    from tempfile import mkstemp, mkdtemp
ImportError: cannot import name 'mkstemp' from partially initialized module 'tempfile' (most likely due to a circular import) (/home/dohlemacher/cloud/ansible/lib/ansible/modules/[tempfile.py](http://tempfile.py/))
```
I found a closed ticket [2] that suggests renaming `modules/tempfile.py`.  I did this.  It fixes my circular dependency.

With this hack of the ansible modules, the module runs:
```
╰─➤  python -m ansible.modules.my_test   /tmp/args.json
{"changed": true, "original_message": "hello", "message": "goodbye", "invocation": {"module_args": {"name": "hello", "new": true}}}
```

[1] https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#verifying-your-module-code-locally
[2] https://github.com/ansible/ansible/issues/25333

### Issue Type

Documentation Report

### Component Name

module hacking

### Ansible Version

```console
╰─➤  ansible --version
ansible 2.10.9
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/dohlemacher/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/dohlemacher/cloud/ansible/lib/ansible
  executable location = /home/dohlemacher/cloud/ansible/bin/ansible
  python version = 3.8.12 (default, Jan 15 2022, 18:39:47) [GCC 7.5.0]

This also happens on the head of devel, but I tried release tag 2.10.9.
```
╰─➤  git show-ref | grep devel
d06d99cbb6a61db043436092663de8b20f6d2017 refs/heads/devel
d06d99cbb6a61db043436092663de8b20f6d2017 refs/remotes/origin/devel
c6d7baeec5bca7f02fc4ddbb7052f8de081f9cdf refs/remotes/origin/temp-2.10-devel
```


### Configuration

```console
$ ansible-config dump --only-changed
ANSIBLE_SSH_ARGS(env: ANSIBLE_SSH_ARGS) =
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False
(END)
```

### OS / Environment
```
╰─➤  cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
```

### Steps to Reproduce

Just follow https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#creating-a-module

### Expected Results
```
╰─➤  python -m ansible.modules.my_test /tmp/args.json

{"changed": true, "original_message": "hello", "message": "goodbye", "invocation": {"module_args": {"name": "hello", "new": true}}}
```
### Actual Results

```console
╰─➤  python -m ansible.modules.my_test   /tmp/args.json
Traceback (most recent call last):
  File "/usr/lib/python3.8/[runpy.py](http://runpy.py/)", line 194, in \_run\_module\_as\_main
    return \_run\_code(code, main\_globals, None,
  File "/usr/lib/python3.8/[runpy.py](http://runpy.py/)", line 87, in \_run\_code
    exec(code, run\_globals)
  File "/home/dohlemacher/cloud/ansible/lib/ansible/modules/[david.py](http://david.py/)", line 72, in \<module>
    from ansible.module\_utils.basic import AnsibleModule
  File "/home/dohlemacher/cloud/ansible/lib/ansible/module\_utils/[basic.py](http://basic.py/)", line 53, in \<module>
    import tempfile
  File "/home/dohlemacher/cloud/ansible/lib/ansible/modules/[tempfile.py](http://tempfile.py/)", line 79, in \<module>
    from tempfile import mkstemp, mkdtemp
ImportError: cannot import name 'mkstemp' from partially initialized module 'tempfile' (most likely due to a circular import) (/home/dohlemacher/cloud/ansible/lib/ansible/modules/[tempfile.py](http://tempfile.py/))
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by bcoca**
fixes #77621

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs

### CODE REVIEW

File: docs/docsite/rst/dev_guide/developing_modules_general.rst [hunk 1]
```diff
@@ -134,11 +99,8 @@ If your module does not need to target a remote host, you can quickly and easily
         }
     }
 
--  If you are using a virtual environment (which is highly recommended for
-   development) activate it: ``$ . venv/bin/activate``
--  Set up the environment for development: ``$ . hacking/env-setup``
--  Run your test module locally and directly:
-   ``$ python -m ansible.modules.my_test /tmp/args.json``
+-  Now yu can run your test module locally and directly:
+   ``$ python -m library/my_test.py /tmp/args.json``
```
Comment (sivel) [inline docs/docsite/rst/dev_guide/developing_modules_general.rst]:
```suggestion
-  Now you can run your test module locally and directly:
   ``$ python library/my_test.py /tmp/args.json``
```

File: docs/docsite/rst/dev_guide/developing_modules_general.rst [hunk 2]
```diff
@@ -121,9 +76,19 @@ Our :ref:`debugging tips <debugging_modules>` will help if you run into bugs as
 Verifying your module code locally
 ----------------------------------
 
+The simplest way is to use ``ansible`` adhoc command:
+
+.. code:: shell
+
+    ansilbe -m library/my_test.py -a '... arguments' remotehost
```
Comment (sqqqrly) [inline docs/docsite/rst/dev_guide/developing_modules_general.rst]: typo in the command

File: docs/docsite/rst/dev_guide/developing_modules_general.rst [hunk 3]
```diff
@@ -121,9 +76,19 @@ Our :ref:`debugging tips <debugging_modules>` will help if you run into bugs as
...
+
 If your module does not need to target a remote host, you can quickly and easily exercise your code locally like this:
 
--  Create an arguments file, a basic JSON config file that passes parameters to your module so that you can run it. Name the arguments file ``/tmp/args.json`` and add the following content:
+.. code:: shell
+
+    ansilbe -m library/my_test.py -a '... arguments' localhost
```
Comment (sqqqrly) [inline docs/docsite/rst/dev_guide/developing_modules_general.rst]: Typo

File: docs/docsite/rst/dev_guide/developing_modules_general.rst [hunk 4]
```diff
@@ -121,9 +76,19 @@ Our :ref:`debugging tips <debugging_modules>` will help if you run into bugs as
...
+
+-  If for any reason you want to avoid going through Ansible, another way is to create an arguments file, a basic JSON config file that passes parameters to your module so that you can run it. Name the arguments file ``/tmp/args.json`` and add the following content:
```
Comment (sqqqrly) [inline docs/docsite/rst/dev_guide/developing_modules_general.rst]: Could list reasons:
- print() is available for debugging.
- pdb works as expected.
- Running the module using Python can be much faster for development.

File: docs/docsite/rst/dev_guide/developing_modules_general.rst [hunk 5]
```diff
@@ -178,34 +146,19 @@ The following two examples will get you started with testing your module code. P
 information, including instructions for :ref:`testing module documentation <testing_module_documentation>`, adding :ref:`integration tests <testing_integration>`, and more.
 
 .. note::
-  Every new module and plugin should have integration tests, even if the tests cannot be run on Ansible CI infrastructure.
+  If contirbuting to Ansible, every new module and plugin should have integration tests, even if the tests cannot be run on Ansible CI infrastructure.
```
Comment (samccann) [inline docs/docsite/rst/dev_guide/developing_modules_general.rst]:
```suggestion
  If contributing to Ansible, every new module and plugin should have integration tests, even if the tests cannot be run on Ansible CI infrastructure.
```

**Review by samccann — APPROVED** [review]

**Pull request merged**